gk/fib.c

```diff
@@ -130,9 +130,20 @@ get_new_ether_cache_locked(struct neighbor_hash_table *neigh,
 	 * to protect the operations here.
 	 */
 	eth_cache->stale = true;
+	eth_cache->l2_len = sizeof(eth_cache->l2_hdr.eth_hdr);
```

Comment:
I think you're right that we do not need to keep track of the L2 length in the Ethernet cache, since we can always tell by the context of where we're using the cache (to send out of the front or the back interface) what the L2 length is.